src/server/v2.0/handler/quota.go

```diff
@@ -0,0 +1,116 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"context"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/goharbor/harbor/src/common/rbac"
+	"github.com/goharbor/harbor/src/controller/quota"
+	"github.com/goharbor/harbor/src/lib"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/server/v2.0/handler/model"
+	"github.com/goharbor/harbor/src/server/v2.0/models"
+	operation "github.com/goharbor/harbor/src/server/v2.0/restapi/operations/quota"
+)
+
+func newQuotaAPI() *quotaAPI {
+	return &quotaAPI{
+		quotaCtl: quota.Ctl,
+	}
+}
+
+type quotaAPI struct {
+	BaseAPI
+	quotaCtl quota.Controller
+}
+
+func (qa *quotaAPI) GetQuota(ctx context.Context, params operation.GetQuotaParams) middleware.Responder {
+	if err := qa.RequireSystemAccess(ctx, rbac.ActionRead, rbac.ResourceQuota); err != nil {
+		return qa.SendError(ctx, err)
+	}
+
+	quota, err := qa.quotaCtl.Get(ctx, params.ID, quota.WithReferenceObject())
+	if err != nil {
+		return qa.SendError(ctx, err)
+
+	}
+	return operation.NewGetQuotaOK().WithPayload(model.NewQuota(quota).ToSwagger(ctx))
+}
+
+func (qa *quotaAPI) ListQuotas(ctx context.Context, params operation.ListQuotasParams) middleware.Responder {
+	if err := qa.RequireSystemAccess(ctx, rbac.ActionList, rbac.ResourceQuota); err != nil {
+		return qa.SendError(ctx, err)
+	}
+
+	query := &q.Query{
+		Keywords: q.KeyWords{
+			"reference":    lib.StringValue(params.Reference),
+			"reference_id": lib.StringValue(params.ReferenceID),
+		},
+		PageNumber: *params.Page,
+		PageSize:   *params.PageSize,
+		Sorting:    lib.StringValue(params.Sort),
```

Comment:
the sorting could be empty here, what about if it's a empty tring?

Reply:
When `Sorting` is an empty string, the DAO of the quota will list by the creation_time in descending order.